### PR TITLE
add some dict functions to IR

### PR DESCRIPTION
**Pull request by catoverdrive**
I removed `toDict` (well, renamed it to `dict`) since it doesn't appear to be used anywhere (the function is in the current function registry as `dict`).

### CODE REVIEW

File: src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala [hunk 1]
```diff
@@ -26,23 +26,68 @@ object DictFunctions extends RegistryFunctions {
...
+    registerIR("keySet", tdict) { d =>
+      val pairs = Ref(genUID(), -types.coerce[TContainer](d.typ).elementType)
+      ToSet(ArrayMap(ToArray(d), pairs.name, GetField(pairs, "key")))
+    }
+
+    registerIR("dict", TSet(TTuple(tv("key"), tv("value"))))(s => ToDict(ToArray(s)))
+
+    registerIR("dict", TArray(TTuple(tv("key"), tv("value"))))(ToDict)
+
+    registerIR("Dict", TArray(tv("key")), TArray(tv("value"))) { (k, v) =>
```
Comment (rcownie) [inline src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: Can you clarify why this one is capitalized as "Dict" when the other conversions are "dict" ?

Comment (tpoterba) [reply src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: this is a relic of the old string expr language, I think. no reason to keep it around.

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: you mean this function, or the naming? if the former, I'll remove it from AST.

Comment (tpoterba) [reply src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: the capitalization, I meant. wasn't clear.

Comment (tpoterba) [reply src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: is this actually used though?

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: I'll check.

Comment (rcownie) [reply src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: I don't have a view either way on whether it should exist.  It just seemed that if the other
conversions are now called lower-case "dict", that should be the name for this one as well.

I guess if the new language doesn't allow it to be called, then it should just disappear.  I
don't know if that's true.

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/ir/functions/DictFunctions.scala]: ok @rcownie I've removed `Dict` from everything since it's not exposed in Python.

**Pull request merged**